src/routes/api/chat/+server.ts

```diff
@@ -0,0 +1,65 @@
+// Defined in a `$REPO/.env` file as described in https://kit.svelte.dev/docs/modules#$env-static-private
+import { CHAT_API_KEY } from '$env/static/private';
+import type { Message, ChatRequest } from '$lib/types';
+
+async function fetchChat(messages: Array<Message>, systemMessage: string = ''): Promise<String> {
+  const ENDPOINT_URL: URL = new URL('https://api.openai.com/v1/chat/completions');
+  const MODEL: string = 'gpt-4o';
+
+  if (systemMessage !== '') {
+    messages.unshift({
+      role: 'system',
+      content: systemMessage
+    });
+  }
+
+  const response = await fetch(ENDPOINT_URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${CHAT_API_KEY}`
+    },
+    body: JSON.stringify({
+      messages: messages,
+      model: MODEL
+      /* Potentially useful arguments later on: (pulled from https://platform.openai.com/docs/api-reference/chat/object)
+      n: 1, // how many completions to generate (i.e. how many "attempts" should it make)
+      max_completion_tokens: null, // how many (max) tokens to generate in total
+      response_format: null, // output format: { "type": "json_schema", "json_schema": {...} } ensures the model will match "json_schema" ;;; { "type" : "json_object" } ensures the model will produce valid json
```

Comment:
Is there a reason we aren't using this for the schema and instead printing typedefs in the prompt?